.github/workflows/ci-java.yml

```diff
@@ -28,7 +28,7 @@ jobs:
         uses: actions/setup-java@v3
         with:
           distribution: 'temurin'
-          java-version: '17'
+          java-version: '11'
```

Comment:
Should we perhaps setup a matrix here and test on both versions, or should we just be testing the lowest version and assuming the latest version is ok? I'm a bit unsure about what the best strategy is here for java.

Reply:
I think having a matrix, consisting of the minimum supported version (v11 with this change) and at least a couple more recent versions (including the most recent version, v17 prior to this change) would be a good idea. The evolution of Java between version 11 and 17 is not something I am familiar with though, so this could be an impossible request. Feedback appreciated!

Reply:
Yea, I think the array of versions makes sense. I think we'd also need to inject that version from the matrix into the `pom.xml` on CI before running it though, right? Maybe other places too?